synapse/http/servlet.py

```diff
@@ -143,45 +143,76 @@ def parse_string(
             is not one of those allowed values.
     """
     return parse_string_from_args(
-        request.args, name, default, required, allowed_values, param_type, encoding
```

Comment:
Removed `param_type` from all of the string parsing functions. It's not used in the codebase and seems superfluous because it's always `string` anyway and just used for the logging.

Reply:
Split out to https://github.com/matrix-org/synapse/pull/10048